src/compas_view2/app/controller.py

```diff
@@ -26,8 +26,128 @@ class Controller:
 
     """
 
-    def __init__(self, app):
+    def __init__(self, app, controller_config: Dict):
         self.app = app
+        self.config = controller_config
+        self.mouse_key = controller_config["actions"]["mouse_key"]
+        self.keys = controller_config["actions"]["keys"]
+        self.mouse = Mouse()
+        self.key_status = {}
+        self.actions = {}  # Currently, no customized actions supported.
+        self.supported_keys = {
+            "shift": QtCore.Qt.Key_Shift,
+            "control": QtCore.Qt.Key_Control,
+            "alt": QtCore.Qt.Key_Alt,
+            "placeholder": None,
+        }
+
+    # ==============================================================================
+    # Providing the basic key and mouse event handling functions.
+    # ==============================================================================
+    def mouse_key_check(self, event, key_status: Dict, mouse_key: Dict):
+        """Giving the mouse key as a dictionary that is in the list of supported keys, return if it is happening (bool).
+
+        Parameters
+        ----------
+        event:
+            The Qt event.
+        key_status : Dict
+            The status dictionary that is like `"key_status": { "shift": False, "control": False, "alt": False }`.
+        mouse_key : Dict
+            The dictionary that is like `"mouse_key": { "mouse": "right", "key": "shift" }`.
+
+        Returns
+        ----------
+        bool:
+            If the mouse key is happening.
+
+
+        Notes
+        ----------
+        This function is designed to check mouse_key combine interactions. The interactions are:
+        "box_selection", "selection", "multi_selection", "deselect", "pan", "rotate", for now.
+        """
+        supported_buttons = {
+            "left": QtCore.Qt.LeftButton,
+            "right": QtCore.Qt.RightButton,
+            "middle": QtCore.Qt.MiddleButton,
+        }
+
+        if event.buttons() & supported_buttons[mouse_key["mouse"]]:
+            if key_status.get(mouse_key["key"]) is None:
+                key_status[mouse_key["key"]] = False
+
+            if mouse_key["key"] == "":
+                for key in key_status.values():
+                    if key:
+                        return False
+                return True
+
+            elif key_status[mouse_key["key"]]:
+                return True
+
+        else:
+            return False
+
+    def mouse_check(self, event, button_name: str):
+        """Giving the button name that is one of the supported buttons, return if the button is pressed.
+
+        Parameters
+        ----------
+        event:
```

Comment:
There needs to be a space after ":", otherwise the sphinx doc will not build correctly

Reply:
Also could you add class for event. same with other occurrences

Reply:
done